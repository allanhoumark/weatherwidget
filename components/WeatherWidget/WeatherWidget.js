import React from 'react';
import Router from 'next/router';
import classNames from 'classnames';
import propTypes from 'prop-types';

class WeatherWidget extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
          city: props.city
        };
        this.handleSubmit = this.handleSubmit.bind(this);
        this.updateCity = this.updateCity.bind(this);
    }

    handleSubmit () {
        event.preventDefault();
        Router.push({
            pathname: '/',
            query:{
                city: this.state.city
            }
        })
    }

    updateCity (evt) {
        this.setState({
          city: evt.target.value
        });
      }

    render() {
        const widgetStyle = {
            style: 'margin: 10px',
            width: '300px'
        }
        
        const panelStyle = classNames('panel','panel-info',{'panel-danger': !this.props.found})
        const {temp, hum, wind, windDirection, city, found}  = this.props;

        return     <div className="widget" style={widgetStyle}>
            <div className={panelStyle}>
            {
                found && <div className="panel-heading">Weather in <b>{city}</b></div>
            }
            {
                !found && <div className="panel-heading">City <b>{city}</b> not found</div>
            }
            <ul className="list-group">
                <li className="list-group-item">Temperature: <b>{temp}Â°C</b></li>
                <li className="list-group-item">Humidity: <b>{hum}</b></li>
                <li className="list-group-item">Wind: <b>{wind} m/s fra {windDirection}</b></li>
                <li className="list-group-item">
                    <form id="weatherwidget" method="GET" onSubmit={this.handleSubmit} className="form-inline">
                    <div className="form-group">
                        <input 
                            value={this.state.city}
                            onChange={evt => this.updateCity(evt)} 
                            type="text" 
                            className="form-control" 
                            id="city" 
                            name="city" 
                            placeholder="City"/>
                    </div>
                    <button type="submit" className="btn btn-default" >Search</button>
                    </form>
                </li>
            </ul>
            
            </div>
        </div>
    }
  }

  WeatherWidget.propTypes = {
      city: propTypes.string.isRequired
  };

export default WeatherWidget