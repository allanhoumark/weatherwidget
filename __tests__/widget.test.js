import React from 'react'; 
import WeatherWidget from '../components/WeatherWidget/WeatherWidget';
import { shallow, mount } from 'enzyme';
import toJSON from 'enzyme-to-json';

const props = {
    city: 'odense',
    temp: 10,
    hum: 75,
    wind: 5,
    windDirection: 'nord',
    found: true
}

describe('<WeatherWidget/>', ()=>{
    it('the widget renders and match snapshot', ()=>{
        const wrapper = shallow(<WeatherWidget {...props}/>);
        expect(toJSON(wrapper)).toMatchSnapshot();
    });
})