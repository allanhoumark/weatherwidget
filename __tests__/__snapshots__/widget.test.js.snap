// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WeatherWidget/> the widget renders and match snapshot 1`] = `
<div
  className="widget"
  style={
    Object {
      "style": "margin: 10px",
      "width": "300px",
    }
  }
>
  <div
    className="panel panel-info"
  >
    <div
      className="panel-heading"
    >
      Weather in 
      <b>
        odense
      </b>
    </div>
    <ul
      className="list-group"
    >
      <li
        className="list-group-item"
      >
        Temperature: 
        <b>
          10
          Â°C
        </b>
      </li>
      <li
        className="list-group-item"
      >
        Humidity: 
        <b>
          75
        </b>
      </li>
      <li
        className="list-group-item"
      >
        Wind: 
        <b>
          5
           m/s fra 
          nord
        </b>
      </li>
      <li
        className="list-group-item"
      >
        <form
          className="form-inline"
          id="weatherwidget"
          method="GET"
          onSubmit={[Function]}
        >
          <div
            className="form-group"
          >
            <input
              className="form-control"
              id="city"
              name="city"
              onChange={[Function]}
              placeholder="City"
              type="text"
              value="odense"
            />
          </div>
          <button
            className="btn btn-default"
            type="submit"
          >
            Search
          </button>
        </form>
      </li>
    </ul>
  </div>
</div>
`;
